torch/nn/attention/__init__.py

```diff
@@ -67,6 +60,32 @@ def _raise_kernel_warnings(params: SDPAParams) -> None:
             can_use_flash_attention(params, True)
 
 
+_backend_names = {
```

Comment:
We could technically move the part in dynamo that uses `_backend_names` over to here instead (get backend name from backend object - needed for a workaround since baking this constant into the FX graph is tricky).

Reply:
I am not too concerned with performance, more just maintainability / readability

Reply:
This is more of a nit btw, I am more just curious if the below changes just tacked on refactoring or the meat of what is required for getting dynamo to not graph break

Reply:
It's a refactor.